### PR TITLE
Implemented: Added missing webhook service endpoints(#85zrgg8kw)

### LINKED ISSUE #232: Add missing webhook service endpoints
## What is the motivation for adding/enhancing this feature?

Add following API endpoint to webhookEndpointUrls:
1. subscribeOrderCreateWebhook
2. subscribeOrderReturnWebhook
3. subscribeOrderCancelWebhook
4. subscribeProductCreateWebhook

https://github.com/hotwax/job-manager/blob/main/src/services/WebhookService.ts#L14

### PR DESCRIPTION
**Pull request by shashwatbangar**
### Related Issues

Closes #232 

### Short Description and Why It's Useful

As we now using general Webhook pub/sub model where we will pass a topic to a single webhook api and according to the topic, API will be called.

Working:
As a normal webhook, we subscribe to a shopifyWebhook and if anything happens int he shopify shop for eg. new orders or cancelled orders, instead of calling an api time to time to check for the status, webhook will be called and will have the results.

### Screenshots of Visual Changes before/after (If There Are Any)

https://user-images.githubusercontent.com/62360781/210044412-83340e31-1618-47ad-a186-82a85217a66c.mov

**IMPORTANT NOTICE** - Remember to add changelog entry

### Contribution and Currently Important Rules Acceptance

- [ ] I read and followed [contribution rules](https://github.com/hotwax/job-manager#contribution-guideline)

### CODE REVIEW

File: src/services/WebhookService.ts [hunk 1]
```diff
@@ -10,25 +10,57 @@ const fetchShopifyWebhooks = async (payload?:  any): Promise <any>  => {
...
+  'ORDER_PAID': {
+    'topic': 'orders/paid',
+    'endpoint': 'orderPaidNotificationFromShopify'
+  }
 } as any
 
 const subscribeWebhook = async (payload?: any, id?: string): Promise <any> => {
-  const endpointUrl = webhookEndpointUrls[id as string];
-  if(!endpointUrl) {
+  const topic = webhookParameters[id as string].topic;
```
Comment (adityasharma7) [inline src/services/WebhookService.ts:53]: Make id in parameters that way we do not have to explicitly cast as string

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/services/WebhookService.ts (hunk 1 above)
Comment (shashwatbangar) [reply src/services/WebhookService.ts]: Done sir

File: src/services/WebhookService.ts [hunk 2]
```diff
@@ -10,25 +10,57 @@ const fetchShopifyWebhooks = async (payload?:  any): Promise <any>  => {
...
+    'topic': 'orders/paid',
+    'endpoint': 'orderPaidNotificationFromShopify'
+  }
 } as any
 
-const subscribeWebhook = async (payload?: any, id?: string): Promise <any> => {
-  const endpointUrl = webhookEndpointUrls[id as string];
-  if(!endpointUrl) {
+const subscribeWebhook = async (payload: any, id: string): Promise <any> => {
+  const topic = webhookParameters[id].topic;
```
Comment (adityasharma7) [inline src/services/WebhookService.ts:53]: Assign webhookParameters[id] to a variable and then use further

File: src/services/WebhookService.ts [hunk 3]
```diff
@@ -10,25 +10,57 @@ const fetchShopifyWebhooks = async (payload?:  any): Promise <any>  => {
...
-  'CANCELLED_ORDERS': '',
-  'PAYMENT_STATUS': '',
-  'RETURNS': '',
-  'NEW_PRODUCTS': '',
-  'DELETE_PRODUCTS': 'service/subscribeProductDeleteWebhook',
-  'BULK_OPERATIONS_FINISH': 'service/subscribeFileStatusUpdateWebhook'
+const webhookParameters = {
+  'NEW_ORDERS': {
+    'topic': 'orders/create',
+    'endpoint': ''
```
Comment (adityasharma7) [inline src/services/WebhookService.ts:16]:
```suggestion
    'endpoint': 'createOrderShopifyWebhook'
```

File: src/services/WebhookService.ts [hunk 4]
```diff
@@ -10,25 +10,57 @@ const fetchShopifyWebhooks = async (payload?:  any): Promise <any>  => {
...
+  'CANCELLED_ORDERS': {
+    'topic': 'orders/cancelled',
+    'endpoint': ''
+  },
+  'PAYMENT_STATUS': {
+    'topic': '',
+    'endpoint': ''
+  },
+  'RETURNS': {
+    'topic': '',
```
Comment (adityasharma7) [inline src/services/WebhookService.ts:27]:
```suggestion
    'topic': 'refunds/create',
```

File: src/services/WebhookService.ts [hunk 5]
```diff
@@ -10,25 +10,57 @@ const fetchShopifyWebhooks = async (payload?:  any): Promise <any>  => {
...
+    'endpoint': ''
```
Comment (adityasharma7) [inline src/services/WebhookService.ts:28]:
```suggestion
    'endpoint': 'returnOrderShopifyWebhook'
```

File: src/services/WebhookService.ts [hunk 6]
```diff
@@ -10,25 +10,57 @@ const fetchShopifyWebhooks = async (payload?:  any): Promise <any>  => {
...
-  'DELETE_PRODUCTS': 'service/subscribeProductDeleteWebhook',
-  'BULK_OPERATIONS_FINISH': 'service/subscribeFileStatusUpdateWebhook'
+const webhookParameters = {
+  'NEW_ORDERS': {
+    'topic': 'orders/create',
+    'endpoint': ''
+  },
+  'CANCELLED_ORDERS': {
+    'topic': 'orders/cancelled',
+    'endpoint': ''
```
Comment (adityasharma7) [inline src/services/WebhookService.ts:20]:
```suggestion
    'endpoint': 'cancelOrderShopifyWebhook'
```

File: src/services/WebhookService.ts [hunk 7]
```diff
@@ -10,25 +10,57 @@ const fetchShopifyWebhooks = async (payload?:  any): Promise <any>  => {
...
+    'topic': '',
+    'endpoint': ''
+  },
+  'RETURNS': {
+    'topic': '',
+    'endpoint': ''
+  },
+  'NEW_PRODUCTS': {
+    'topic': 'products/create',
+    'endpoint': ''
```
Comment (adityasharma7) [inline src/services/WebhookService.ts:32]:
```suggestion
    'endpoint': 'createProductShopifyWebhook'
```

Comment (adityasharma7) [inline src/services/WebhookService.ts:32]:
```suggestion
    'endpoint': 'createProductShopifyWebhook'
```

File: src/services/WebhookService.ts [hunk 8]
```diff
@@ -10,25 +10,57 @@ const fetchShopifyWebhooks = async (payload?:  any): Promise <any>  => {
...
+    'topic': 'orders/create',
+    'endpoint': ''
+  },
+  'CANCELLED_ORDERS': {
+    'topic': 'orders/cancelled',
+    'endpoint': ''
+  },
+  'PAYMENT_STATUS': {
+    'topic': '',
+    'endpoint': ''
```
Comment (adityasharma7) [inline src/services/WebhookService.ts:24]:
```suggestion
    'endpoint': 'orderPaidNotificationFromShopify'
```

File: src/services/WebhookService.ts [hunk 9]
```diff
@@ -10,25 +10,57 @@ const fetchShopifyWebhooks = async (payload?:  any): Promise <any>  => {
...
+  'NEW_ORDERS': {
+    'topic': 'orders/create',
+    'endpoint': ''
+  },
+  'CANCELLED_ORDERS': {
+    'topic': 'orders/cancelled',
+    'endpoint': ''
+  },
+  'PAYMENT_STATUS': {
+    'topic': '',
```
Comment (adityasharma7) [inline src/services/WebhookService.ts:23]:
```suggestion
    'topic': 'orders/paid',
```

File: src/services/WebhookService.ts [hunk 10]
```diff
@@ -10,25 +10,57 @@ const fetchShopifyWebhooks = async (payload?:  any): Promise <any>  => {
...
-  if(!endpointUrl) {
+const subscribeWebhook = async (payload: any, id: string): Promise <any> => {
+  const topic = webhookParameters[id].topic;
+  const endpoint = webhookParameters[id].endpoint;
+  if(!endpoint) {
     showToast(translate("Configuration missing"));
     return;
   }
+  payload['topic'] = topic;
+  payload['endpoint'] = endpoint;
```
Comment (adityasharma7) [inline src/services/WebhookService.ts:60]: `<OMS URL> + '/shopify/' + endpoint`

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/services/WebhookService.ts (hunk 2 above)
Comment (shashwatbangar) [reply src/services/WebhookService.ts:53]: Done sir

File: src/services/WebhookService.ts (hunk 3 above)
Comment (shashwatbangar) [reply src/services/WebhookService.ts:16]: Ok

File: src/services/WebhookService.ts (hunk 4 above)
Comment (shashwatbangar) [reply src/services/WebhookService.ts:27]: Ok

File: src/services/WebhookService.ts (hunk 5 above)
Comment (shashwatbangar) [reply src/services/WebhookService.ts:28]: Ok

File: src/services/WebhookService.ts (hunk 6 above)
Comment (shashwatbangar) [reply src/services/WebhookService.ts:20]: Ok

File: src/services/WebhookService.ts (hunk 7 above)
Comment (shashwatbangar) [reply src/services/WebhookService.ts:32]: Ok

Comment (shashwatbangar) [reply src/services/WebhookService.ts:32]: Ok

File: src/services/WebhookService.ts (hunk 8 above)
Comment (shashwatbangar) [reply src/services/WebhookService.ts:24]: Ok

File: src/services/WebhookService.ts (hunk 9 above)
Comment (shashwatbangar) [reply src/services/WebhookService.ts:23]: Ok

File: src/services/WebhookService.ts (hunk 10 above)
Comment (shashwatbangar) [reply src/services/WebhookService.ts:60]: Got it sir

**Comment by k2maan** [comment]
Looks good to me.

File: src/services/WebhookService.ts [hunk 11]
```diff
@@ -10,25 +11,61 @@ const fetchShopifyWebhooks = async (payload?:  any): Promise <any>  => {
...
+
+  const webhookParameter = webhookParameters[id]
+  const topic = webhookParameter.topic;
+  const endpoint = webhookParameter.endpoint;
+  if(!endpoint) {
     showToast(translate("Configuration missing"));
     return;
   }
+  payload['topic'] = topic;
+  payload['endpoint'] = baseURL + 'shopify/' + endpoint;
```
Comment (adityasharma7) [inline src/services/WebhookService.ts:65]: `<OMS URL without /api> + 'shopify/' + endpoint` 
https://github.com/hotwax/job-manager/blob/main/src/views/Settings.vue#L181

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/services/WebhookService.ts (hunk 11 above)
Comment (shashwatbangar) [reply src/services/WebhookService.ts:65]: Done sir

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**